api/plugin/httpconnector/httpconnectorhost.go

```diff
@@ -0,0 +1,98 @@
+package httpconnector
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"time"
+
+	apiplugin "github.com/iansmith/parigot/api/plugin"
+	pcontext "github.com/iansmith/parigot/context"
+	"github.com/iansmith/parigot/eng"
+	"github.com/iansmith/parigot/g/httpconnector/v1"
+	"github.com/tetratelabs/wazero/api"
+	"google.golang.org/protobuf/proto"
+)
+
+const port = ":9000"
+
+type HttpConnectorPlugin struct{}
+
+var httpCnt *httpConnectorImpl
+
+type httpConnectorImpl struct {
+	ctx      context.Context
+	httpChan chan *http.Request
+}
+
+func (*HttpConnectorPlugin) Init(ctx context.Context, e eng.Engine) bool {
+	e.AddSupportedFunc(ctx, "httpconnector", "check_", checkHost)
+
+	httpCnt = newHttpCntSvc(ctx)
+	go runHttpListener(httpCnt)
+
+	return true
+}
+
+func newHttpCntSvc(ctx context.Context) *httpConnectorImpl {
+	newCtx := pcontext.ServerGoContext(ctx)
+
+	c := &httpConnectorImpl{
+		ctx:      newCtx,
+		httpChan: make(chan *http.Request),
+	}
+
+	return c
+}
+
+// runHttpListener starts an HTTP listener on port 9000 and listens for incoming requests from outside.
+// When a request is received, it is sent to the httpChan channel of the provided httpConnectorImpl instance.
+func runHttpListener(hCnt *httpConnectorImpl) {
+	h1 := func(w http.ResponseWriter, req *http.Request) {
+		if req != nil {
+			hCnt.httpChan <- req
+		}
+		io.WriteString(w, "start running HTTP listener!")
+	}
+	http.HandleFunc("/", h1)
+
+	http.ListenAndServe(port, nil)
+}
+
+func hostBase[T proto.Message, U proto.Message](ctx context.Context, fnName string,
```

Comment:
did you add this just for the error recovery? I think this function already exists in api/plugin/syscall...

Reply:
Do you mean I could remove the `hostBase` function and call `apiplugin.InvokeImplFromStack` directly?